### PR TITLE
chore: deprecate Angular 13 and below

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
We are currently on Angular 12. 

Used the **Angular migration tool** (https://update.angular.io/?l=3&v=12.0-14.0) to deprecate Angular 13 and below, and jump to Angular 14 on
- `ui-angular` package
- example angular app
- canary angular app

#### Description of how you validated changes
- Unit tests
- Tested on external sample apps built with Angular v14, v15 and v16 to use `@aws-amplify/ui-angular` (upgraded to Angular v14).
- e2e tests against example app

Docs:
- Update Angular docs to fix style imports and polyfill setup.

SCR-20230531-kxfw

SCR-20230531-kxko

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/angular/src/polyfills.ts [hunk 1]
```diff
@@ -18,18 +18,6 @@
  * BROWSER POLYFILLS
  */
 
-/**
- * IE11 requires the following for NgClass support on SVG elements
- */
-// import 'classlist.js';  // Run `npm install --save classlist.js`.
```
Comment (wlee221) [inline examples/angular/src/polyfills.ts:24]: 😈

File: examples/angular/angular.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
             "polyfills": "src/polyfills.ts",
             "tsConfig": "tsconfig.app.json",
             "assets": ["src/favicon.ico", "src/assets"],
-            "styles": ["src/styles.scss"],
+            "styles": ["src/styles.scss", "@aws-amplify/ui-angular/theme.css"],
```
Comment (wlee221) [inline examples/angular/angular.json:44]: I know I wrote this 🤣 , but I think we should put `@aws-amplify/ui-angular/theme.css` first so that app styles win over our styles.

File: docs/src/pages/[platform]/getting-started/installation/styles.angular.mdx [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import { Tabs, TabItem } from '@aws-amplify/ui-react';
 
+Open your `angular.json` file, and add `node_modules/@aws-amplify/ui-angular/theme.css` to `styles` array in your `angular.json`. This array is located in `projects.<project-name>.architect.build.options`.
+
 ```js
-/* ./src/styles.css */
-@import '~@aws-amplify/ui-angular/theme.css';
+"styles": [
+  "src/styles.css",
+  "node_modules/@aws-amplify/ui-angular/theme.css"
```
Comment (wlee221) [inline docs/src/pages/[platform]/getting-started/installation/styles.angular.mdx:8]: Same nit as below, let's put `ui-angular` styles above `src/styles.css`.

File: .changeset/itchy-eggs-film.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+"@aws-amplify/ui-docs": patch
+"@aws-amplify/ui-angular-example": patch
+"@aws-amplify/ui-angular-mono": patch
+"@aws-amplify/ui-angular": patch
```
Comment (wlee221) [inline .changeset/itchy-eggs-film.md:5]:
```suggestion
"@aws-amplify/ui-angular": major
```

File: .changeset/itchy-eggs-film.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+---
+
+chore: deprecate Angular 13 or below
```
Comment (wlee221) [inline .changeset/itchy-eggs-film.md:8]: let's make this changeset scarier, maybe something like

```suggestion
**Breaking**: bump minimum Angular version to 14. If your application is on Angular 13 or older, please migrate to Angular 14+ as per [official guide](https://update.angular.io/).
```

File: .changeset/itchy-eggs-film.md (hunk 4 above)
Comment (sreeramsama) [reply .changeset/itchy-eggs-film.md:5]: It would get upgraded to major when `@aws-amplify/ui` is set to `major`, right?

Comment (wlee221) [reply .changeset/itchy-eggs-film.md:5]: yes, but this change itself also deserves a major bump on Angular. This would ensure changelog gets formatted as expected.

Comment (wlee221) [reply .changeset/itchy-eggs-film.md:5]: Changeset will combine the bumps intelligently when it's time to release.

Comment (wlee221) [reply .changeset/itchy-eggs-film.md:5]: I think we want to add a new changeset for each breaking changes PR, since each breaking change PRs affect different packages.

**Review by wlee221 — APPROVED** [review]
Approving prematurely for the technical portion. I'll err on others for migration docs changes once the PR is updated.

**Comment by reesscot** [comment]
Can you please add a step in the Installation instructions linking to the Global Polyfill steps? Customers have been confused by this, there are github issues that Troubleshooting is hard to find.

File: packages/angular/jest.config.js [hunk 6]
```diff
@@ -18,16 +18,16 @@ module.exports = {
   testPathIgnorePatterns: ['<rootDir>/node_modules/', '<rootDir>/dist/'],
   transformIgnorePatterns: ['node_modules/(?!.*.(mjs|ts)$)'],
   moduleNameMapper: {
-    uuid: require.resolve('uuid'),
-    nanoid: require.resolve('nanoid'),
+    '^uuid$': '<rootDir>/../../node_modules/uuid',
+    '^nanoid$': '<rootDir>/../../node_modules/nanoid',
```
Comment (calebpollman) [inline packages/angular/jest.config.js]: Updated to align to the syntax used in existing _jest.config.js_ files

File: packages/angular/jest.config.js [hunk 7]
```diff
@@ -18,16 +18,16 @@ module.exports = {
...
-      tsconfig: '<rootDir>/tsconfig.spec.json',
-      stringifyContentPathRegex: '\\.(html|svg)$',
-    },
+    '^.+\\.{ts|tsx}?$': [
+      'ts-jest',
+      {
+        tsconfig: '<rootDir>/tsconfig.spec.json',
+        stringifyContentPathRegex: '\\.(html|svg)$',
+      },
+    ],
```
Comment (calebpollman) [inline packages/angular/jest.config.js:31]: Fixes deprecation warnings

**Review by wlee221 — APPROVED** [review]
LGTM

File: canary/apps/angular/angularcli/angular.json [hunk 8]
```diff
@@ -19,7 +19,7 @@
             "polyfills": "src/polyfills.ts",
             "tsConfig": "tsconfig.app.json",
             "assets": ["src/favicon.ico", "src/assets"],
-            "styles": ["src/styles.css"],
+            "styles": ["@aws-amplify/ui-angular/theme.css", "src/styles.css"],
```
Comment (reesscot) [inline canary/apps/angular/angularcli/angular.json:22]: It's weird to me that Angular Authenticator has it's own styling. Do we have a ticket to track removing this?

Comment (calebpollman) [reply canary/apps/angular/angularcli/angular.json:22]: +1

File: canary/apps/angular/angularcli/src/styles.css [hunk 9]
```diff
@@ -1,3 +1 @@
 /* You can add global styles to this file, and also import other style files */
-
-@import '~@aws-amplify/ui-angular/theme.css';
```
Comment (calebpollman) [inline canary/apps/angular/angularcli/src/styles.css:3]: Do we need this file still?

File: packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap [hunk 10]
```diff
@@ -34,7 +33,7 @@ exports[`amplify-error renders as expected 1`] = `
       <button
         amplify-button=""
         aria-label="Dismiss alert"
-        class="amplify-button amplify-button--link amplify-button--medium"
+        class="amplify-field-group__control amplify-alert__dismiss amplify-button amplify-button--link amplify-button--medium"
```
Comment (calebpollman) [inline packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: Why is this necessary?

**Review by calebpollman — COMMENTED** [review]
Mostly LGTM, left a couple questions in the PR. Additionally looking for clarification on the reasoning/need for the styling changes to the `Authenticator` since this PR is scoped to updating the Angular version. I understand some of it is just alphabetizing, but it looks like we have added additional classnames to some of the HTML elements as well?

File: canary/apps/angular/angularcli/angular.json (hunk 8 above)
Comment (wlee221) [reply canary/apps/angular/angularcli/angular.json:22]: I believe this is just the reexport of `@aws-amplify/ui/styles.css`, that this canary app needs to import.

File: canary/apps/angular/angularcli/src/styles.css (hunk 9 above)
Comment (wlee221) [reply canary/apps/angular/angularcli/src/styles.css:3]: It could go either way, but since this is a default autogenerated file, I’m fine with leaving it as is.

File: packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap (hunk 10 above)
Comment (sreeramsama) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: The earlier version of Angular package did not properly resolve those extra classnames for buttons in dismissible alerts.
This change fixes that snapshot.

Comment (reesscot) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: What does it mean that the earlier version didn't resolve the classnames? Were they always added by jest didn't render them to a snapshot correctly, is that what you mean? Is this related to upgrading jest and testing-library?

File: canary/apps/angular/angularcli/angular.json (hunk 8 above)
Comment (reesscot) [reply canary/apps/angular/angularcli/angular.json:22]: I created a ticket

File: packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap (hunk 10 above)
Comment (calebpollman) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: > This change fixes that snapshot.

Seems like it would change the rendered UI as well?

Comment (wlee221) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: So I looked into a little bit into this. So previously, we had an ambiguous code:

https://github.com/aws-amplify/amplify-ui/blob/41a7e90f4572d1d7e28d2d17d3be6bf42125bbac/packages/angular/projects/ui-angular/src/lib/primitives/error/error.component.html#L23-L24

`amplify-button` directive adds its own classnames (`amplify-button`, `amplify-button--link`, and `amplify-button--medium`).

And then we tried to append two additional classnames (`amplify-field-group__control`, `amplify-alert__dismiss`). 

Now the two versions (12, and 14) treat this code differently. In 12, they do not append the two new classnames. In 14, they merge the classnames.

Comment (calebpollman) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: Thanks for looking in this @wlee221

Comment (sreeramsama) [reply packages/angular/projects/ui-angular/src/lib/primitives/error/__tests__/__snapshots__/error.component.test.ts.snap:36]: Makes sense. Will remove classnames `amplify-field-group__control` and `amplify-alert__dismiss` to be consistent with Angular 12.

File: docs/src/pages/[platform]/getting-started/installation/index.page.mdx [hunk 11]
```diff
@@ -27,6 +27,10 @@ export async function getStaticProps() {
   {({ platform }) => import(`./dependencies.${platform}.mdx`)}
 </Fragment>
 
+<Fragment platforms={['web']} useCommonWebContent>
+  {({ platform }) => import(`./shims.${platform}.mdx`)}
+</Fragment>
+
```
Comment (wlee221) [inline docs/src/pages/[platform]/getting-started/installation/index.page.mdx:33]: Found a docs bug:

image

It tries to load `shim.<platform>.mdx` for all frameworks. Can we fix this and scope this section only to Angular?

Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/installation/index.page.mdx]: Added fix to update `shims.angular.mdx` and remove `shims.web.mdx`

File: packages/angular/projects/ui-angular/src/lib/primitives/error/error.component.html [hunk 12]
```diff
@@ -21,7 +21,6 @@
   </div>
   <button
     amplify-button
-    class="amplify-field-group__control amplify-alert__dismiss"
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives/error/error.component.html:24]: FYI for context: https://github.com/aws-amplify/amplify-ui/pull/3861/files

I created a ticket to handle classname change in another PR.

**Review by wlee221 — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
